lib/runtime/wasmer/exports.go

```diff
@@ -86,6 +87,55 @@ func (in *Instance) GrandpaAuthorities() ([]types.Authority, error) {
 	return types.GrandpaAuthoritiesRawToAuthorities(gar)
 }
 
+// BabeGenerateKeyOwnershipProof returns the babe key ownership proof from the runtime.
+func (in *Instance) BabeGenerateKeyOwnershipProof(slot uint64, authorityID [32]byte) (
+	types.OpaqueKeyOwnershipProof, error) {
+
+	// scale encoded slot uint64 + scale encoded array of 32 bytes
+	const maxBufferLength = 8 + 33
+	buffer := bytes.NewBuffer(make([]byte, 0, maxBufferLength))
+	encoder := scale.NewEncoder(buffer)
+	err := encoder.Encode(slot)
+	if err != nil {
+		return nil, fmt.Errorf("encoding slot: %w", err)
+	}
+	err = encoder.Encode(authorityID)
+	if err != nil {
+		return nil, fmt.Errorf("encoding authority id: %w", err)
+	}
+
+	encodedKeyOwnershipProof, err := in.Exec(runtime.BabeAPIGenerateKeyOwnershipProof, buffer.Bytes())
+	if err != nil {
+		return nil, fmt.Errorf("executing %s: %w", runtime.BabeAPIGenerateKeyOwnershipProof, err)
```

Comment:
It would be nice to have this error case covered in a unit test (that is, if invalid slot+authority id is possible)

Reply:
doesn't seem like it, I have tried this with empty authority id and the call doesn't throw error